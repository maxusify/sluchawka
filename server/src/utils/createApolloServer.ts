import { ApolloServerPluginLandingPageDisabled, ApolloServerPluginLandingPageGraphQLPlayground } from 'apollo-server-core';
import { ApolloServer } from 'apollo-server-express';
import { ApolloContext } from 'src/types';
import { buildSchema } from 'type-graphql';

import { FindUniqueUserResolver } from '../../prisma/generated/type-graphql';
import { UserResolver } from '../resolvers/User';
import { createPrismaClient } from './createPrismaClient';
import { createRedisClient } from './createRedisClient';

const createApolloServer = async () => {
  return new ApolloServer({
    schema: await buildSchema({
      resolvers: [UserResolver, FindUniqueUserResolver],
      validate: false,
    }),
    context: ({ req, res }): ApolloContext => ({
      prisma: createPrismaClient(),
      req,
      res,
      redis: createRedisClient(),
    }),
    plugins: [
      ApolloServerPluginLandingPageGraphQLPlayground(),
      ApolloServerPluginLandingPageDisabled(),
    ],
  });
};

export default createApolloServer;