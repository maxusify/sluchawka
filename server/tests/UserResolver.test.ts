import { ApolloServer, ExpressContext } from "apollo-server-express";
import "reflect-metadata";

import createApolloServer from "../src/utils/createApolloServer";

let server: ApolloServer<ExpressContext>;

beforeAll(async () => {
  server = await createApolloServer();
});

describe("User Resolver", () => {
  it("Creates new user correctly", async () => {
    const email = `test@test.com`;
    const password = "12345678";

    const QUERY_DATA = `
      mutation($data: UserAuthInput!) {
        register(data: $data) {
          errors {
            field,
            message
          }
          user {
            email,
          }
        }
      }
    `;

    const data = {
      data: {
        email: email,
        password: password,
      },
    };

    const response = await server.executeOperation({
      query: QUERY_DATA,
      variables: data,
    });

    expect(response.errors).toBeUndefined();
    expect(response).toMatchSnapshot();
  });

  it("Log in new user correctly", async () => {
    const email = `test@test.com`;
    const password = "12345678";

    const QUERY_DATA = `
      mutation Login($data: UserAuthInput!) {
        login(data: $data) {
          errors {
            field,
            message
          }
          user {
            id,
            email,
            createdAt,
            updatedAt
          }
        }
      }
    `;

    const data = {
      data: {
        email: email,
        password: password,
      },
    };

    const response = await server.executeOperation({
      query: QUERY_DATA,
      variables: data,
    });

    expect(response.errors).toBeUndefined();
    expect(response).toMatchSnapshot();
  });
});

afterAll(async () => {
  server.stop();
});
