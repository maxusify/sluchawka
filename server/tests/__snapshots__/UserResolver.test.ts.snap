// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Resolver Creates new user correctly 1`] = `
Object {
  "data": Object {
    "register": Object {
      "errors": Array [
        Object {
          "field": "email",
          "message": "Provided email address is already in use.",
        },
      ],
      "user": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`User Resolver Log in new user correctly 1`] = `
Object {
  "data": Object {
    "login": Object {
      "errors": null,
      "user": Object {
        "createdAt": "2022-06-30T14:05:48.619Z",
        "email": "test@test.com",
        "id": "57c2889f-1eb9-47cf-82d8-9d181ae229cf",
        "updatedAt": "2022-06-30T14:05:48.620Z",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
